### PR TITLE
Device get neighbour devices

### PR DESCRIPTION
**Pull request by marcin-usielski**

### CODE REVIEW

File: moler/device/textualdevice.py [hunk 1]
```diff
@@ -90,20 +93,76 @@ def __init__(self, sm_params=None, name=None, io_connection=None, io_type=None,
...
+            self.name,
+            self.__class__.__module__,
+            self.__class__.__name__,
+            self._state_prompts
+        )
+        self._open = True
         self._log(level=logging.INFO, msg=msg)
-        self._log(level=logging.INFO, msg=self._state_prompts)
+
+    def is_open(self):
```
Comment (Ernold11) [inline moler/device/textualdevice.py]: Maybe `def is_established` is better name? In this place we check that connection to device is established,

File: moler/device/textualdevice.py [hunk 2]
```diff
@@ -90,20 +93,76 @@ def __init__(self, sm_params=None, name=None, io_connection=None, io_type=None,
...
+            neighbour_device.add_neighbour_device(neighbour_device=self, bidirectional=False)
+
+    def get_neighbour_devices(self, device_type):
+        """
+        Returns list of neighbour devices of passed type.
+
+        :param device_type: type of device. If None then all neighbour devices will be returned.
+        :return: list of devices.
+        """
+        f_devices = list()
```
Comment (Ernold11) [inline moler/device/textualdevice.py]: old name -> rename to neighbour_device

File: test/device/test_device.py [hunk 3]
```diff
@@ -14,6 +14,47 @@ def test_device_directly_created_must_be_given_io_connection(buffer_connection):
     assert dev.io_connection == buffer_connection
 
 
+def test_device_add_neighbour_device(buffer_connection):
+    from moler.device.unixlocal import UnixLocal
+
+    dev1 = UnixLocal(io_connection=buffer_connection)
+    dev2 = UnixLocal(io_connection=buffer_connection)
+    f_devices = dev1.get_neighbour_devices(device_type=UnixLocal)
```
Comment (Ernold11) [inline test/device/test_device.py]: change all f_devices to neighbour_devices

File: moler/device/textualdevice.py [hunk 4]
```diff
@@ -90,20 +93,76 @@ def __init__(self, sm_params=None, name=None, io_connection=None, io_type=None,
...
-                                                                                      self.__class__.__module__,
-                                                                                      self.__class__.__name__)
+
+        msg = "Established connection to device '{}' (as instance of class '{}.{}') with prompts: '{}'.".format(
+            self.name,
+            self.__class__.__module__,
+            self.__class__.__name__,
+            self._state_prompts
+        )
+        self._open = True
```
[the remark this answers is not in the document]
Comment (marcin-usielski) [reply moler/device/textualdevice.py]: Done

File: moler/device/textualdevice.py (hunk 2 above)
Comment (marcin-usielski) [reply moler/device/textualdevice.py]: Done

File: test/device/test_device.py (hunk 3 above)
Comment (marcin-usielski) [reply test/device/test_device.py]: Done

**Pull request merged**